userland/src/parameters.cpp

```diff
@@ -0,0 +1,74 @@
+#include <algorithm>
+#include <chrono>
+#include <cmath>
+#include <iostream>
+#include <string>
+#include <thread>
+
+#include <rclcpp/rclcpp.hpp>
+
+template <typename ParameterTypeT>
+ParameterTypeT
+wait_for_future(rclcpp::node::Node::SharedPtr & node, std::shared_future<ParameterTypeT> & future)
+{
+  std::future_status status;
+  do {
+    rclcpp::spin_some(node);
+    status = future.wait_for(std::chrono::milliseconds(100));
```

Comment:
The reason for waiting is that you need to wait to know the future's status (ready, deferred, timeout). Here the future here is waiting for a very short time so as to block spinning the least as possible. I can make the timeout even shorter, though.